OPA Helm chart migration

angelbarrera92 opened the pull request:
Hello OPA team!

As you know, the official helm chart repository is now deprecated, and vendors have to find a place where to publish and distribute their helm charts. 

This is the reason why we post here: https://github.com/helm/charts/pull/8915#issuecomment-728050130 to help your team hosting the OPA helm chart in another location. Thanks to @tsandall, he created a team granting access to this repository.

## TL;DR
```bash
$ helm repo add open-policy-agent https://open-policy-agent.github.io/kube-mgmt/charts
$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "open-policy-agent" chart repository
Update Complete. ⎈Happy Helming!⎈
$ helm search repo opa
NAME                    CHART VERSION   APP VERSION     DESCRIPTION                                       
open-policy-agent/opa   1.14.6          0.15.1          Open source, general-purpose policy engine. Enf...
```

What we did in this PR was to:

- Create the same chart structure you are currently using in the `gatekeeper` repository.
  - Copy the OPA Helm Chart code from `github.com/helm/charts/stable` to the new location in this repo.
  - Remove the deprecation notes.
- Copy and adapt the CI of the `gatekeeper` repository (GitHub actions) to publish the chart under GitHub Pages.
  - We ignore everything from `gatekeeper` CI but the release of the Helm Chart.
- Configure the repository to use the `gh-pages` branch as the source of GitHub pages.

As you can see, everything is "working," but for sure, it can be improved:

- Be consistent in the CI (Travis or GitHub actions)
- Include missing steps in the GitHub actions CI
- Update the values of the Helm Chart with the newest versions
- E2E tests of the helm chart
- ...

I wanted to keep this PR as small as possible. (Enough to migrate the helm chart).
Feedback is welcomed!

phisco commented:
I would have preferred using Travis and uploading the charts with a release, but if this is the same approach gatekeeper is using I guess it's fine.

@tsandall, if this is ok with you I think we can merge it.

tsandall commented:
LGTM. Do we have to create a tag to test the chart publishing? RE: Travis-vs-GHA: we want to migrate off Travis completely so GHA is fine (#78).

angelbarrera92 commented:
> LGTM. Do we have to create a tag to test the chart publishing? RE: Travis-vs-GHA: we want to migrate off Travis completely, so GHA is fine (#78).

Ciao @tsandall 

I've tested it hardcoding the name of the branch (reverted in the last commit of this PR: https://github.com/open-policy-agent/kube-mgmt/pull/79/commits/99c595420bc6c2f344dc755357c2a60c8b6e2ac2)

I've configured it to follow the gatekeeper repository's behavior, so creating a new tag (starting by a `v`) triggers the GHA pipeline.

Thanks!

phisco commented on charts/opa/Chart.yaml:
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+appVersion: 0.15.1
+description: Open source, general-purpose policy engine. Enforce fine-grained invariants over arbitrary Kubernetes resources.
+name: opa
```
I think the name "opa" does make sense as the chart is deploying the most purely "opa" thing you could deploy on Kubernetes (with only the addition of kube-mgmt). Moreover, that's what people are used to.

phisco commented on charts/opa/README.md:
````diff
@@ -0,0 +1,108 @@
+configuration.
+
+```bash
+$ helm repo add open-policy-agent https://open-policy-agent.github.io/kube-mgmt/charts
````
It might make sense, but in the end it's just a suggestion, so I don't know.

angelbarrera92 replied:
As @phisco said, it is just an example. Maybe we can make it shorter like:

`helm repo add opa https://open-policy-agent.github.io/kube-mgmt/charts`

phisco commented on .github/workflows/opa-helm-chart-release.yaml:
```diff
@@ -0,0 +1,34 @@
+        uses: actions/checkout@v2
+
+      - name: Publish Helm chart
+        uses: stefanprodan/helm-gh-pages@v1.2.0
```
Totally agree with this, we'll work on it.

angelbarrera92 replied:
Addressed here: https://github.com/open-policy-agent/kube-mgmt/pull/79#issuecomment-734322656

phisco commented:
@angelbarrera92 we still miss to port the github action to a simpler and local script, if you have time to work on it feel free to, otherwise I'll try do have it done by the end of this week

angelbarrera92 commented:
https://github.com/open-policy-agent/kube-mgmt/pull/79#discussion_r526386758

> @angelbarrera92 we still miss to port the github action to a simpler and local script, if you have time to work on it feel free to, otherwise I'll try do have it done by the end of this week

Addressed. What I did was to copy-paste the script from the GitHub action (I've modified just a line to add `sudo` to the installation of helm binary in the `/usr/local/bin/helm` path) into the `build` directory. 

This way it will be easier to update the script just in case the GitHub action is updated.

phisco reviewed (approved):
LGTM, let's wait for pipelines to finish and merge it

bueti commented:
What's the state on this PR?

phisco commented:
merging 👍
